apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -227,38 +233,51 @@ external_binary_to_mess_id(BExtMessID) when is_binary(BExtMessID) ->
 %% -----------------------------------------------------------------------
 %% XML
 
+%% @doc Adds all arcid elements (for backward compatibility)
```

Comment:
I'm not sure we need to add the old `archived` and `stanza-id`. From what I remember the `archived` was only in MAMv02 and `stanza-id` should replace it when this XEP was created. It was only clearly stated in MAM 0.6 that it has to be added, before it was optional.